c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -223,7 +231,325 @@ struct Block {
   bool is_split() const {
     return (prev != nullptr) || (next != nullptr);
   }
+  void splice(Block* before, Block* after) {
+    if (before) {
+      before->next = this;
+    }
+    prev = before;
+    if (after) {
+      after->prev = this;
+    }
+    next = after;
+  }
+};
+
+#define C10_CUDA_DRIVER_CHECK(EXPR)                   \
+  do {                                                \
+    CUresult __err = EXPR;                            \
+    if (__err != CUDA_SUCCESS) {                      \
+      const char* err_str;                            \
+      CUresult get_error_str_err C10_UNUSED =         \
+          cuGetErrorString(__err, &err_str);          \
+      if (get_error_str_err != CUDA_SUCCESS) {        \
+        AT_ERROR("CUDA driver error: unknown error"); \
+      } else {                                        \
+        AT_ERROR("CUDA driver error: ", err_str);     \
+      }                                               \
+    }                                                 \
+  } while (0)
+
+#if !defined(USE_ROCM) && !defined(USE_BAZEL) && !defined(FBCODE_CAFFE2) && \
```

Comment:
any particular reason for the `FBCODE_CAFFE2` check

Reply:
There is a build rule change (link libtorch_cuda.so to libcuda.so) and so there will need to be an internal equivalent change. I want to figure out those internal build changes in a separate PR.